data_steward/tools/generate_rids_from_participantsummaryAPI.py

```diff
@@ -0,0 +1,256 @@
+
+The current participants should be obtained using the Participant Summary API.
+The data cutoff data for the Oct 2020 CDR is 8/1/2020.
+
+"""
+# Python imports
+import argparse
+from datetime import datetime, date
+
+# Third party imports
+from google.cloud import bigquery
+import pandas as pd
+import numpy as np
+import pandas_gbq
+
+# Project imports
+from utils.participant_summary_requests import get_access_token, get_participant_data
+from utils.bq import JINJA_ENV
+
+GET_EXISTING_RIDS = JINJA_ENV.from_string("""
+SELECT DISTINCT research_id FROM `{{project}}.{{dataset}}.{{table}}`
+""")
+
+GET_PIDS_IN_EXISITNG_MAPPING = JINJA_ENV.from_string("""
+SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.{{table}}`
+""")
+
+
+def calculate_max_age_date(max_age):
+    """
+    Helper method to get the current date to pass to the participant summary API based on the max_age
+    :param max_age: max age of participants
+    :return: date x years ago based on max age
+    """
+    today = date.today()
+    year = today.year - int(max_age)
+    return date(year, today.month, today.day)
+
+
+def milliseconds_since_epoch():
+    """
+    Helper method to get the number of milliseconds from the epoch to now
+    :return:  an integer number of milliseconds
+    """
+    return int(
+        (datetime.utcnow() - datetime(1970, 1, 1)).total_seconds() * 1000)
+
+
+def get_participants(start_date, end_date, max_age, existing_pids):
+    """
+    Method to hit the participant summary API based on cutoff dates and max age. Filters out participants that already
+    exist in the pipeline_tables.pid_rid_mapping table.
+
+    Loops through the results to double check the API returned the correct participants.
+
+    :param start_date: start date of cutoff date range
+    :param end_date: end date of cutoff date range
+    :param max_age: max age of participants that were initially excluded, that now need to be backfilled
+    :param existing_pids: list of pids that already exist in mapping table
+    :return: dataframe with single column person_id from participant summary API, which needs RIDS created for
+    """
+    # get max age date to pass as parameter to Participant Summary API
+    max_age_date = calculate_max_age_date(max_age)
+
+    # create datetime from cutoff_date string
+    end_date_obj = datetime.strptime(end_date, '%Y-%m-%dT%H:%M:%S')
+    start_date_obj = datetime.strptime(start_date, '%Y-%m-%dT%H:%M:%S')
+
+    # Make request to get API version. This is the current RDR version for reference
+    # See https://github.com/all-of-us/raw-data-repository/blob/master/opsdataAPI.md for documentation of this api.
+    url_cutoff_participants = "https://{0}.appspot.com/rdr/v1/ParticipantSummary?_sort=lastModified" \
+                              "&withdrawalStatus={1}&consentForStudyEnrollmentAuthored=gt{2}" \
+                              "&consentForStudyEnrollmentAuthored=lt{3}".format('all-of-us-rdr-prod', 'NOT_WITHDRAWN',
```

Comment:
RDR project needs to be parameterized. we should not be putting these project names in public.